src/generalizedlinearmixedmodel.jl

```diff
@@ -529,6 +534,52 @@ ranef(m::GeneralizedLinearMixedModel; uscale::Bool=false) = ranef(m.LMM, uscale=
 
 LinearAlgebra.rank(m::GeneralizedLinearMixedModel) = first(m.LMM.feterms).rank
 
+"""
+    refit!(m::GeneralizedLinearMixedModel[, y::Vector];
+          fast::Bool = (length(m.θ) == length(m.optsum.final)), 
+          nAGQ::Integer = m.optsum.nAGQ))
+
+Refit the model `m` after installing response `y`.
+
+If `y` is omitted the current response vector is used.
+
+If not specified, the `fast` and `nAGQ` options from the previous fit are used.
+
+"""
+function refit!(m::GeneralizedLinearMixedModel{T};
+                fast::Bool = (length(m.θ) == length(m.optsum.final)), 
+                nAGQ::Integer = m.optsum.nAGQ)  where T
+    
+    deviance!(m, 1)
+    reevaluateAend!(m.LMM)
+    
+    reterms = m.LMM.reterms
+    optsum = m.LMM.optsum
+    # we need to reset optsum so that it
+    # plays nice with the modifications fit!() does 
+    optsum.lowerbd = foldl(vcat, lowerbd(c) for c in reterms)
```

Comment:
```suggestion
    optsum.lowerbd = mapfoldl(lowerbd, vcat, reterms)
```
Can you write it this way instead of using a generator?